### DISCUSSION TITLE
Ultralytics for Finance | Recognition and Prediction of Trading Patterns

### DISCUSSION THREAD

**Discussion by abstractumx** [ultralytics/ultralytics#9507] [Ideas]
Hi there, 

can you apply your algo features for understanding and predict trading patterns? (industry field: FINANCE)

Do you have already some use cases?

**Comment by glenn-jocher** [comment]
Hi! 👋

Indeed, our YOLO models, including YOLOv8, can be tailored for a wide range of applications, including in the finance sector for recognizing and predicting trading patterns. While we don't have specific use cases in finance publicly available, the flexibility and accuracy of YOLO models mean they can potentially be adapted for financial analysis, given the right dataset for training.

For recognizing trading patterns, you’d need a dataset of financial charts annotated with the patterns you're interested in. Then, you could train a YOLO model to detect these patterns. Here's a very simplified example of how you might start training on your custom dataset:

```bash
yolo train data=your_finance_dataset.yaml model=yolov8n.yaml
```

As for prediction, while YOLO models excel at detection and recognition tasks, predicting future trading patterns might involve time-series analysis and might require integrating YOLO's outputs with other prediction models designed for financial forecasting.

Hope this helps! Let us know if you have any more questions.

**Reply by abstractumx** [reply]
does YOLO works only with yaml files?

**Reply by glenn-jocher** [reply]
@abstractumx hello! 😊

YOLO models primarily use YAML files for configuration (e.g., dataset specification, model architecture), ensuring a structured and easy-to-edit setup. However, for tasks like training, validation, prediction, or export, you can directly use images, videos, directories, or even streams as input without needing YAML files. Here's a quick example for detection:

```bash
yolo detect predict model=yolov8n.pt source='path/to/image.jpg'
```

This allows you to get started with predictions right away! YAML files come into play for more structured tasks like training with specific datasets. Hope that clears it up!
